lib/web_ui/lib/src/engine/skwasm/skwasm_impl/image.dart

```diff
@@ -55,8 +56,26 @@ class SkwasmImage extends SkwasmObjectWrapper<RawImage> implements ui.Image {
 
   @override
   Future<ByteData?> toByteData(
-      {ui.ImageByteFormat format = ui.ImageByteFormat.rawRgba}) {
-    return (renderer as SkwasmRenderer).surface.rasterizeImage(this, format);
+      {ui.ImageByteFormat format = ui.ImageByteFormat.rawRgba}) async {
+    if (format == ui.ImageByteFormat.png) {
+      final ui.PictureRecorder recorder = ui.PictureRecorder();
+      final ui.Canvas canvas = ui.Canvas(recorder);
+      canvas.drawImage(this, ui.Offset.zero, ui.Paint());
+      final DomImageBitmap bitmap =
+        await (renderer as SkwasmRenderer).surface.renderPicture(
+          recorder.endRecording() as SkwasmPicture,
+        );
+      final DomOffscreenCanvas offscreenCanvas =
+        createDomOffscreenCanvas(bitmap.width.toDartInt, bitmap.height.toDartInt);
+      final DomCanvasRenderingContextBitmapRenderer context =
+        offscreenCanvas.getContext('bitmaprenderer')! as DomCanvasRenderingContextBitmapRenderer;
+      context.transferFromImageBitmap(bitmap);
+      final DomBlob blob = await offscreenCanvas.convertToBlob();
+      final JSArrayBuffer arrayBuffer = (await blob.arrayBuffer().toDart)! as JSArrayBuffer;
```

Comment:
Once we have the bytes of the image, should we set `offscreenCanvas.height/width` to zero so the browser can eagerly reclaim the bitmap memory? Otherwise, the browser may have to wait for the GC cycle to kick in. In the past, Safari would run out of memory if we didn't zero out the canvas, and return `null` when creating a new canvas.

Reply:
Good idea. I'll do that